### PR TITLE
fix: filter out empty env vars

### PR DESCRIPTION
**Pull request by edewit**
fixes: #1101

### CODE REVIEW

File: packages/launcher-component/src/loaders/buildimage-loader.tsx [hunk 1]
```diff
@@ -41,7 +41,9 @@ const convertToPairs = object => {
 export const convertToObject = (vars: Array<{ key: string; value: string }>) => {
   const result = {};
   for (const env of vars) {
-    result[env.key] = env.value;
```
Comment (ia3andy) [inline packages/launcher-component/src/loaders/buildimage-loader.tsx]: shouldn't we merge the other one (2d array) before fixing this 🤔😁

Comment (edewit) [reply packages/launcher-component/src/loaders/buildimage-loader.tsx]: resolved by rebasing ;)

**Comment by ia3andy** [comment]
@edewit can we close?

**Comment by edewit** [comment]
@ia3andy no we still need to filter out the empty keys, I've rebased and fixed it to use the `string[][]` version of the component

File: packages/launcher-component/src/loaders/buildimage-loader.tsx [hunk 2]
```diff
@@ -30,7 +30,12 @@ const convertToPairs = (object?: {[key: string]: string}): string[][] => {
 };
 
 export const convertToObject = (vars: string[][]) => {
-  return _.fromPairs(vars);
+  return _.fromPairs(vars.reduce<string[][]>((result, p) => {
```
Comment (ia3andy) [inline packages/launcher-component/src/loaders/buildimage-loader.tsx]: is it not possible to use `.filter` instead of `.reduce`?

Comment (edewit) [reply packages/launcher-component/src/loaders/buildimage-loader.tsx]: if you use filter instead of reduce you'll get `undefined` as an property name

**Pull request merged**